client/my-sites/themes/controller.jsx

```diff
@@ -107,14 +107,17 @@ export function fetchThemeData( context, next, shouldUseCache = false ) {
 		page: 1,
 		number: DEFAULT_THEME_QUERY.number,
 	};
-	const cacheKey = context.path;
+	// context.pathname includes tier, filter, and verticals, but not the query string, so it's a suitable cacheKey
+	// However, we can't guarantee it's normalized -- filters can be in any order, resulting in multiple possible cacheKeys for
```

Comment:
> However, we can't guarantee it's normalized

Not for this PR, but probably for M6 we should make the [validate-filters middleware](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/themes/validate-filters.js) isomorphic so we can redirect to canonical route on the server.